task19/js/slider.js

```diff
@@ -0,0 +1,169 @@
+  }
+  slideImageList.appendChild(slideImageFragment);
+};
+
+const renderSlideButton = (slideImageData) => {
+  const directions = ["previous", "next"];
+  directions.forEach((direction) => {
+    const buttonElement = createElementWithClassName("button", "slide-button");
+    const buttonImage = document.createElement("img");
+    buttonElement.id = `js-${direction}Button`;
+    buttonImage.src = `./img/${direction}-arrow.png`;
+    buttonElement.append(buttonImage);
+    direction === "previous" && slideShowWrapper.before(buttonElement);
+    direction === "next" && slideShowWrapper.after(buttonElement);
+  });
+  addEventForButton(slideImageData);
+};
+
+const renderPagination = (slideImageData) => {
+  const paginationListElement = createElementWithClassName("ul", "pagination-list");
+  const paginationFragment = document.createDocumentFragment();
+  for (let i = 0; i < slideImageData.length; i++) {
+    const paginationItemElement = createElementWithClassName("li", "pagination-item");
+    paginationItemElement.dataset.index = i;
+    i === 0 && paginationItemElement.classList.add("is-show");
+    paginationFragment.appendChild(paginationListElement).appendChild(paginationItemElement);
+  }
+  slideShowContainer.appendChild(paginationFragment);
+  addEventForPagination(slideImageData);
+};
+
+const renderSlideNumber = (slideImageData) => {
+  const slideNumberTextElement = createElementWithClassName("p", "slide-number");
+  const slideLength = slideImageData.length;
+  slideNumberTextElement.id = "js-slideNumber";
+  slideNumberTextElement.textContent = `${currentImageIndex + 1} / ${slideLength}`;
+  slideShowContainer.appendChild(slideNumberTextElement);
+};
+
+const changeSlider = (slideImageData) => {
+  changeCurrentNumber(slideImageData);
+  toggleDisabledOfButton(slideImageData);
+  changeImage();
+  changePagination();
+};
+
+const changeCurrentNumber = (slideImageData) => {
+  document.getElementById("js-slideNumber").textContent = `${currentImageIndex + 1} / ${slideImageData.length}`;
+}
+
+const toggleDisabledOfButton = (slideImageData) => {
+  const nextButton = document.getElementById("js-nextButton");
+  const previousButton = document.getElementById("js-previousButton");
+  const firstSlideImage = 0;
+  const lastSlideImage = slideImageData.length - 1;
+  nextButton.disabled = currentImageIndex === lastSlideImage;
+  previousButton.disabled = currentImageIndex === firstSlideImage;
+};
+
+const changeImage = () => {
+  const slideImage = document.querySelectorAll(".slide-image-item");
+  document.querySelector(".is-active").classList.remove("is-active");
+  slideImage[currentImageIndex].classList.add("is-active");
+}
+
+const changePagination = () => {
+  const paginations = document.querySelectorAll(".pagination-item");
+  document.querySelector(".is-show").classList.remove("is-show");
+  paginations[currentImageIndex].classList.add("is-show");
+}
```

Comment:
IMO
As the behaviour of both functions is exactly the same, it is better to combine them into one function.
By unifying the is-show and is-active class names, the functions can be combined into one.

[CodeSandBox](https://codesandbox.io/s/damp-tdd-2ibq7w?file=/js/slider.js)

```suggestion
//同じ動きの関数をまとめる
const changeActiveItem = (item) => {
  const items = document.querySelectorAll(item);
  document.querySelector(`${item}.is-show`).classList.remove("is-show");
  items[currentImageIndex].classList.add("is-show");
};
```

The caller specifies the element.
In this case, 
the class name given to the first element and the CSS class name must also be changed.

````JavaScript
//呼び出し元で要素を指定する
const changeSlider = (slideImageData) => {
  changeCurrentNumber(slideImageData);
  toggleDisabledOfButton(slideImageData);
  changeActiveItem(".slide-image-item"); //ここで要素を指定
  changeActiveItem(".pagination-item"); //ここで要素を指定
};
````

Reply:
Thanks for sharing your great ideas!!!
Very easy to read.Very helpful sharing of CodeSandBox.
[eb957c5](https://github.com/seeeen044/JavaScript_handson/pull/20/commits/eb957c5798430bcfba0f1912c250298e151cac2b)